# Ćwiczenie 1: Series — tworzenie i operacje (20 min)

# Krok 1 — Series z listy i dict


In [2]:
import pandas as pd
import numpy as np
# Series z listy — domyślny indeks (0, 1, 2, ...)
oceny = pd.Series([4.5, 3.0, 5.0, 3.5, 4.0])
print(oceny)

0    4.5
1    3.0
2    5.0
3    3.5
4    4.0
dtype: float64


Zadanie 1a: Utwórz Series pensje z dict:

In [6]:
# Klucze: 'Anna','Jan','Ewa','Marek','Kasia'
# Wartości: 5500, 7200, 4800, 9100, 6300
slownik = dict(zip( ['Anna','Jan','Ewa','Marek','Kasia'],[5500, 7200, 4800, 9100, 6300]))
pensje = pd.Series(slownik)
print(pensje)

Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64


Zadanie 1b: Odpowiedz na pytania o pensje :


In [ ]:
# 1. Wyświetl imię osoby zarabiającej najwięcej (użyj idxmax())
print(pensje.idxmax())
# 2. Oblicz średnią pensję
print(pensje.mean())
# 3. Wyświetl imiona osób zarabiających powyżej średniej (filtrowanie boolean)
print(pensje[pensje>pensje.mean()])
# 4. Policz ile osób zarabia poniżej 6000 (użyj len() lub .sum())
print(len(pensje[pensje<6000]))

Marek
6580.0
Jan      7200
Marek    9100
dtype: int64
2


# Krok 2 — Operacje na Series

Zadanie 1c: Oblicz pensje po zmianach:

In [26]:
# 1. Oblicz pensje po podwyżce 10% dla wszystkich (pensje * 1.10)
print(pensje*1.10)
# 2. Oblicz pensje po premii 500 zł dla wszystkich (pensje + 500)
print(pensje+500)
# 3. Oblicz nową pensję = (pensja * 1.10) + 500
new_pensje = pensje * 1.10 + 500
print(new_pensje)
# 4. Wyświetl ranking  nowych pensji od najwyższej (użyj sort_values)
print(new_pensje.sort_values(ascending=0))

Anna      6050.0
Jan       7920.0
Ewa       5280.0
Marek    10010.0
Kasia     6930.0
dtype: float64
Anna     6000.0
Jan      7700.0
Ewa      5300.0
Marek    9600.0
Kasia    6800.0
dtype: float64
Anna      6550.0
Jan       8420.0
Ewa       5780.0
Marek    10510.0
Kasia     7430.0
dtype: float64
Marek    10510.0
Jan       8420.0
Kasia     7430.0
Anna      6550.0
Ewa       5780.0
dtype: float64


# Ćwiczenie 2: DataFrame — tworzenie i atrybuty (20 min)

# Krok 1 — DataFrame z dict

In [28]:
# Dane o produktach sklepu internetowego
sklep = pd.DataFrame({
'produkt': ['Laptop','Tablet','Smartfon','Słuchawki','Monitor'],
'cena': [3500, 1800, 2500, 350, 2200],
'stan_magazynowy': [45, 120, 200, 500, 75],
'kategoria': ['Komputery','Mobilne','Mobilne','Akcesoria','Komputery']
})



Zadanie 2a: Zbadaj DataFrame sklep:

In [40]:
# 1. Wyświetl kształt tabeli (shape)
print(sklep.shape)
# 2. Wyświetl nazwy kolumn (columns)
print(sklep.columns)
# 3. Wyświetl typy danych w każdej kolumnie (dtypes)
print(sklep.dtypes)
# 4. Wyświetl pierwsze 3 wiersze (head)
sklep.head(3)

(5, 6)
Index(['produkt', 'cena', 'stan_magazynowy', 'kategoria', 'wartosc_magazynu',
       'cena_z_vat'],
      dtype='str')
produkt                 str
cena                  int64
stan_magazynowy       int64
kategoria               str
wartosc_magazynu      int64
cena_z_vat          float64
dtype: object


,produkt,cena,stan_magazynowy,kategoria,wartosc_magazynu,cena_z_vat
0,Laptop,3500,45,Komputery,157500,4305.0
1,Tablet,1800,120,Mobilne,216000,2214.0
2,Smartfon,2500,200,Mobilne,500000,3075.0


Zadanie 2b: Dodaj nowe kolumny do DataFrame sklep:

In [37]:
# 1. Dodaj kolumnę 'wartosc_magazynu'= cena * stan_magazynowy
sklep['wartosc_magazynu'] = sklep.cena * sklep.stan_magazynowy
# 2. Dodaj kolumnę 'cena_z_vat' = cena * 1.23
sklep['cena_z_vat'] = sklep.cena * 1.23
# 3. Wyświetl cały DataFrame z nowymi kolumnami
sklep.info()

<class 'pandas.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   produkt           5 non-null      str    
 1   cena              5 non-null      int64  
 2   stan_magazynowy   5 non-null      int64  
 3   kategoria         5 non-null      str    
 4   wartosc_magazynu  5 non-null      int64  
 5   cena_z_vat        5 non-null      float64
dtypes: float64(1), int64(3), str(2)
memory usage: 372.0 bytes


# Krok 2 — Wczytanie CSV

In [41]:
# Wczytaj dataset tips
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url)

Zadanie 2c: Podstawowa eksploracja datasetu tips:

In [46]:
# 1. Wyświetl ile wierszy i kolumn ma dataset (shape)
print(tips.shape)
# 2. Wyświetl ostatnie 5 wierszy (tail)
print(tips.tail(5))
# 3. Wyświetl 3 losowe wiersze (sample)
print(tips.sample(3))
# 4. Wyświetl typy danych w każdej kolumnie (dtypes)
print(tips.dtypes)

(244, 7)
     total_bill   tip     sex smoker   day    time  size
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2
     total_bill   tip     sex smoker   day    time  size
221       13.42  3.48  Female    Yes   Fri   Lunch     2
133       12.26  2.00  Female     No  Thur   Lunch     2
93        16.32  4.30  Female    Yes   Fri  Dinner     2
total_bill    float64
tip           float64
sex               str
smoker            str
day               str
time              str
size            int64
dtype: object


# Ćwiczenie 3: Eksploracja datasetu — samodzielna analiza (30 min)

Dane

In [47]:
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)

Zadanie 3a: Pierwszy rzut oka

In [57]:
# 1. Wyświetl head(10)
print(penguins.head(10))
# 2. Wyświetl shape
print(penguins.shape)
# 3. Uruchom info() — ile brakujących wartości widzisz?
penguins.info()
print(f"Brakujące dane:\n{penguins.isna().sum()}")

  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       36

Zadanie 3b: Statystyki opisowe

In [64]:
# 1. Uruchom describe() — co mówią ci średnie i odchylenia?
# że najbardziej różni się masa ciała
print(penguins.describe())
# 2. Uruchom describe(include='all') — co dodają kolumny tekstowe?
#ile unikalnych, ile mody, modę, uzwględnia wszystkie kolumny
print(penguins.describe(include='all'))
# 3. Jaka jest mediana masy ciała (body_mass_g)?
print(penguins.body_mass_g.mean())


       bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000
       species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count      344     344      342.000000     342.000000         342.000000   
unique       3       3             NaN            NaN                NaN   
top     Adelie  Biscoe             NaN            NaN                NaN   
freq       152     168             NaN            NaN                NaN   

Zadanie 3c: Rozkłady wartości

In [84]:
# 1. Ile jest pingwinów każdego gatunku? (value_counts na 'species')
print(penguins.value_counts('species'))
# 2. Ile pingwinów na każdej wyspie? (value_counts na 'island')
print(penguins.value_counts('island'))
# 3. Rozkład płci (value_counts na 'sex') — czy jest zbalansowany?
print(penguins.value_counts('sex')) #różnica 3
# 4. Jaki procent to gatunek Adelie? (normalize=True)
print(f"{(penguins.value_counts('species',normalize=True)['Adelie']*100):.2f}%")
#44.186%

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64
island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64
sex
MALE      168
FEMALE    165
Name: count, dtype: int64
44.19%


Zadanie 3d: Brakujące dane

In [80]:
# 1. Policz brakujące wartości w każdej kolumnie (isna().sum())
print(penguins.isna().sum())
# 2. Jaki procent danych brakuje w kolumnie 'sex'?
print(f"Procent danych któryh brakuje w kolumnie sex: {(penguins['sex'].isna().sum()/len(penguins)*100):.2f}%")

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64
Procent danych któryh brakuje w kolumnie sex: 3.20%


Zadanie 3e: Wnioski (refleksja) W komórce Markdown napisz minimum 5 obserwaci o datasecie. Każda obserwacja to jedno zdanie poparte liczbą z twoich obliczeń powyżej. 

Przykłady:


Dataset zawiera 342 pingwinów z 3 gatunków...
Najczęstszy gatunek to Adelie (152 osobników, 44.20%)"
Brakuje danych w kolumnach bill_length_mm, bill_depth_mm,flipper_length_mm,body_mass_g,sex(łącznie 19 braków)
Średnia masa ciała to 4201.754386 g (mediana: 4050 g)"
Pingwiny mieszkają na 3 wyspach, najliczniej na Biscoe

# Ćwiczenie 4: Pytania biznesowe + commit (15 min)

# Dane — wróć do datasetu tips

In [86]:
tips = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv')
# Dodaj kolumnę z procentem napiwku
tips['tip_pct'] = (tips['tip'] / tips['total_bill'] * 100).round(1)

Zadanie 4a:

In [106]:
# 1. Oblicz średni rachunek (total_bill)
print(tips.total_bill.mean())
# 2. Oblicz średni napiwek w procentach (tip_pct)
print(tips.tip_pct.mean())
# 3. Wyświetl który dzień ma najwięcej rachunków (value_counts na 'day')
print(tips.value_counts('day').idxmax())
# 4. Policz ile rachunków jest od palaczy (value_counts na 'smoker')
print(tips.value_counts('smoker')['Yes'])
# 5. Wyświetl describe() tylko dla kolumny 'tip'— jaka jest mediana napiwku?
print(tips['tip'].describe()['mean'])
# 6. Jaki procent rachunków pochodzi z obiadu (Dinner)?  
print(f"Procent rachunków którzy pochodzą z obiadu (Dinner): {(tips['time'].value_counts()['Dinner']/len(tips)*100):.2f}%")
# 7. Ile unikalnych rozmiarów grup jest w danych? (nunique na 'size')
print(tips['size'].nunique())

19.78594262295082
16.08155737704918
Sat
93
2.99827868852459
Procent rachunków którzy pochodzą z obiadu (Dinner): 72.13%
6
